types/google-spreadsheet/index.d.ts

```diff
@@ -0,0 +1,1154 @@
+     * @description
+     * A1 column letter of the last column in the worksheet
+     */
+    lastColumnLetter: string;
+
+    /**
+     * @description
+     * stats about the cells in the worksheet
+     */
+    cellStats: CellStats;
+
+    // #endregion
+
+    // #region SYNCHRONOUS METHODS
+
+    /**
+     * @description
+     * retrieve a cell from the cache based on numeric row and column indices
+     * - zero-based index
+     */
+
+    getCell(rowIndex: number, columnIndex: number): GoogleSpreadsheetCell;
+
+    /**
+     * @description
+     * retrieve a cell from the cache based on A1 address
+     */
+    getCellByA1(a1Address: string): GoogleSpreadsheetCell;
+
+    /**
+     * @description
+     * fetch all rows in the worksheet
+     *
+     * @param options pagination options
+     */
+    getRows(options?: PaginationOptions): Promise<GoogleSpreadsheetRow[]>;
+
+    /**
+     * @description
+     * reset local cache of properties and cell data
+     * - cache is empties so props and cells must be re-fetched
+     *
+     * @param dataOnly if true, only affects data, not properties
+     */
+    resetLocalCache(dataOnly: boolean): void;
+
+    // #endregion
+
+    // #region ASYNCHRONOUS METHODS
+
+    /**
+     * @description
+     * loads the cells in the worksheet
+     * - note: if using an API key (read-only access), only A1 ranges are supported
+     * @param filters single or array of filters
+     * - strings are treated as an A1 range
+     * - objects are treated as a WorksheetGridRange
+     */
+    loadCells(filters: string | WorksheetGridRange | string[] | WorksheetGridRange[]): Promise<void>;
+
+    /**
+     * @description
+     * saves all cells in the worksheet that have unsaved changes
+     */
+    saveUpdatedCells(): Promise<void>;
+
+    /**
+     * @description
+     * saves all cells that have unsaved changes
+     *
+     * @param cells array of cells to save
+     */
+    saveCells(cells: GoogleSpreadsheetCell[]): Promise<void>;
+
+    /**
+     * @description
+     * set the header (first) row in the worksheet
+     *
+     * @param headers
+     */
+    setHeaderRow(headers: string[]): Promise<void>;
+
+    /**
+     * @description
+     * loads the header row (first row) of the sheet
+     * - usually do not need to call this directly
+     */
+    loadHeaderRow(): Promise<void>;
+
+    /**
+     * @description
+     * append a row to the end of the worksheet
+     *
+     * @param values row values as either:
+     * - an object of header and value pairs (relative to the worksheet header columns)
+     * - an array of values in column order
+     * @param options insertion options
+     * - raw: ?DESCRIPTION?
+     * - insert:?DESCRIPTION?
+     */
+    addRow(
+        values:
+            | {
+                  [header: string]: string | number | boolean;
+              }
+            | Array<string | number | boolean>,
+        options?: { raw: boolean; insert: boolean },
+    ): Promise<void>;
```

Comment:
There is a clash with the original source file and return here:
https://github.com/theoephraim/node-google-spreadsheet/blob/abb8217c2d9b6fc1da731ec9dfeda3f97b349747/lib/GoogleSpreadsheetWorksheet.js#L403-L405
The method cannot be Promise<void>, as it returns `rows[0]`, probably:
```suggestion
    ): Promise<GoogleSpreadsheetRow>;
```